light_curves/lc_classifier.ipynb

```diff
@@ -0,0 +1,1029 @@
+    "            df_norm = pd.concat([df_norm, dfsingle])\n",
+    "            \n",
+    "            \n",
+    "#reset the index\n",
+    "df_norm = df_norm.set_index([\"objectid\", \"label\", \"band\", \"datetime\"])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "#try standardizing each object separately\n",
+    "#same idea that we are trying to remove the signal that some objects are brighter than others\n",
+    "#but keep the info about relative differences within each object.\n",
+    "\n",
+    "#setup new column to hold the normalized fluxes\n",
+    "df_reindex[[\"normflux\"]] = 100\n",
+    "\n",
+    "#can't figure out how to do this when using multiindex, so get rid of it for a minute\n",
+    "#remove the index\n",
+    "df_reindex_reset = df_reindex.reset_index()\n",
+    "\n",
+    "\n",
+    "#how many objects are in the dataframe?\n",
+    "#actually this is the max and some of them are missing\n",
+    "nobj = df_reindex.index.unique('objectid').max() + 1\n",
+    "\n",
+    "#for each objct in the dataframe\n",
+    "for oid in tqdm(range(nobj)):\n",
+    "    try:\n",
+    "        #what is the median flux of all bands for this object?\n",
+    "        meanflux = df_reindex.loc[oid,:,:,:].flux.mean(skipna = True)\n",
+    "        stdflux = df_reindex.loc[oid,:,:,:].flux.std(skipna=True)\n",
+    "        print('mean flux', meanflux)\n",
+    "    except KeyError:\n",
+    "        #some of the data cleaning above must have removed some objects\n",
+    "        print(oid, ' objectid doesnt exist')\n",
+    "    else:\n",
+    "        querystring = f'objectid == {oid}'\n",
+    "        zflux = (df_reindex_reset.query(querystring).flux.values - meanflux) / stdflux\n",
+    "        datetime = df_reindex_reset.query(querystring).datetime.values\n",
+    "        objectid = df_reindex_reset.query(querystring).objectid.values\n",
+    "        band = df_reindex_reset.query(querystring).band.values\n",
+    "        label = df_reindex_reset.query(querystring).label.values\n",
+    "        dfsingle = pd.DataFrame(dict(flux=zflux, err=df_reindex.loc[oid,:,:,:].err, datetime=datetime, objectid=objectid, band=band,label=label))\n",
+    "\n",
+    "        #now concatenate the single new df with df_lc\n",
+    "        if oid == 0:\n",
+    "            df_norm = dfsingle\n",
+    "        else:\n",
+    "            df_norm = pd.concat([df_norm, dfsingle])\n",
+    "            \n",
+    "            \n",
+    "#reset the index\n",
+    "df_norm = df_norm.set_index([\"objectid\", \"label\", \"band\", \"datetime\"])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#proove this did what I wanted it to do for one randomly chosen object\n",
+    "oid = 318\n",
+    "meanflux = df_norm.loc[oid,:,:,:].flux.mean(skipna = True)\n",
```

Comment:
I actually like keeping the extra `:, :, :`.  It seems more clear to me to keep them so that the structure of the thing is apparent to anyone trying to figure out what I did (or me at any time > 5 min after I wrote it).  Unless you are truly opposed, I'd like to keep them.

Reply:
I'm not opposed and I agree that it's more clear especially in the context of this notebook where there are lots of other spots where `:` is required.